landingzones/lz-machinelearning/lz.bicep

```diff
@@ -336,14 +379,14 @@ param hubNetwork object
 @description('Network configuration.  Includes peerToHubVirtualNetwork flag, useRemoteGateway flag, name, dnsServers, addressPrefixes and subnets (oz, paz, rz, hrz, privateEndpoints, sqlmi, databricksPublic, databricksPrivate, aks) ')
 param network object
 
-var sqldbPassword = sqldb.enabled ? '${uniqueString(rgStorage.id)}*${toUpper(uniqueString(sqldb.username))}' : ''
+var sqldbPassword = sqldb.enabled && !sqldb.aadAuthenticationOnly  ? '${uniqueString(rgStorage.id)}*${toUpper(uniqueString(sqldb.sqlAuthenticationUsername))}' : ''
 var sqlmiPassword = sqlmi.enabled ? '${uniqueString(rgStorage.id)}*${toUpper(uniqueString(sqlmi.username))}' : ''
 
 var databricksName = 'databricks'
 var databricksEgressLbName = 'egressLb'
 var datalakeStorageName = 'datalake${uniqueString(rgStorage.id)}'
-var amlMetaStorageName = 'amlmeta${uniqueString(rgCompute.id)}'
-var akvName = 'akv${uniqueString(rgSecurity.id)}'
+var amlMetaStorageName = 'amlmeta${uniqueString(rgSecurity.id)}'
+var akvName = 'akv${uniqueString(rgCompute.id)}'
```

Comment:
The unique string should be based on rgSecurity.id since Key Vault will be deployed to Security RG.